### DISCUSSION TITLE
Unable to install Foundry-zkSync? Could be because of libusb

### DISCUSSION THREAD

**Discussion by spencerj411** [Cyfrin/foundry-full-course-cu#2157] [General]
Hi, Cyfrin Updraft fam! In case someone else is experiencing the same issue, I experienced difficulty installing foundry-zksync. I followed the quick install steps here: https://github.com/matter-labs/foundry-zksync?tab=readme-ov-file#quick-install. I ran the command from step 3. I then got the following output:

```
############################################################################################ 100.0%
foundryup-zksync: downloading manpages
############################################################################################ 100.0%
dyld[47517]: Library not loaded: /opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib
  Referenced from: <4E22E9D2-C2E1-3E64-91FC-74FA51F97C20> /Users/darramos/.foundry/bin/forge (built for macOS 14.5 which is newer than running OS)
  Reason: tried: '/opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib' (no such file), '/opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib' (no such file)
foundryup-zksync: command failed: /Users/darramos/.foundry/bin/forge --version
foundryup-zksync: installed -
dyld[47520]: Library not loaded: /opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib
  Referenced from: <EFAE5C47-A562-397F-AD0F-71FC137A87C3> /Users/darramos/.foundry/bin/cast (built for macOS 14.5 which is newer than running OS)
  Reason: tried: '/opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib' (no such file), '/opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib' (no such file)
foundryup-zksync: command failed: /Users/darramos/.foundry/bin/cast --version
foundryup-zksync: installed -
foundryup-zksync: done!
Verifying installation...
dyld[47524]: Library not loaded: /opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib
  Referenced from: <4E22E9D2-C2E1-3E64-91FC-74FA51F97C20> /Users/darramos/.foundry/bin/forge (built for macOS 14.5 which is newer than running OS)
  Reason: tried: '/opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib' (no such file), '/opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib' (no such file)
Installation verification failed. Forge is not properly installed.
```

The cause of it is because I did not have libusb installed on my homebrew. I was able to fix the issue by running `brew install libusb`. When I ran step 3 again, it installed foundry-zksync successfully. I am posting this as I didn't see a post about libusb being a dependency. Hope this helps someone!

**Comment by EngrPips** [comment]
Awesome, Thanks for having the interest of fellow students at heart. We appreciate you.

**Comment by Stachelabs** [comment]
I have the same issue but running ```brew install libusb```didn't help much. Any guess? Can I skip all this zksync thing otherwise? thanks

**Reply by EngrPips** [reply]
Hello @silent-pooler, The `zksync` part of the course is optional, and you can skip it if it causes you too much headache. working with `zksync` is not generally smooth as of now, and we expect that to change with time.

**Reply by Stachelabs** [reply]
thanks, exactly what I was thinking about ahah

**Reply by EngrPips** [reply]
Yeah, That is what it is lol.

**Reply by Keivou** [reply]
Hey @EngrPips I see you're in all the latest discussions on this issue, your input is much appreciated :)

**Reply by EngrPips** [reply]
Anytime, Friend.

**Comment by surfrestadev** [comment]
Ran into the same issue and this thread solved it for me. Posting here because the thread is a bit old, but the solution worked for me just now.

**Reply by EngrPips** [reply]
Awesome, I am glad the suggested solution worked for you.
